sdk/servicebus/azure-messaging-servicebus/README.md

```diff
@@ -399,9 +399,24 @@ ServiceBusReceiverClient receiver = new ServiceBusClientBuilder()
 receiver.close();
 ```
 
+### When to use 'ServiceBusProcessorClient'.
+
+When to use 'ServiceBusProcessorClient', 'ServiceBusReceiverClient' or ServiceBusReceiverAsyncClient? 
+ 
+The best option for receiving messages in most common cases is 'ServiceBusProcessorClient'. The processor can handle errors automatically and is designed to receive messages continuously. The processor has a simple API to set concurrency for processing messages in parallel.
+
+The low-level client, 'ServiceBusReceiverAsyncClient', is for advanced users who want more control and flexibility over their Reactive application at the expense of more complexity in the application. Unlike the processor, the low-level 'ServiceBusReceiverAsyncClient' does not have automatic recovery built into it. The reactive application using this client needs to deal with terminal events and choose the operators (to add recovery, manage backpressure, threading) in the Reactor chain. As stated earlier, for common asynchronous receive situations, the 'ServiceBusProcessorClient' should be the first choice.
+
+'ServiceBusReceiverClient' usage is discouraged unless you want to have pull semantics or want to migrate existing code base that uses synchronous receiver approach. For high-throughput and parallel message processing use cases, use 'ServiceBusProcessorClient'.
+
 ### Sharing of connection between clients
-The creation of physical connection to Service Bus requires resources. An application should share the connection  
-between clients which can be achieved by sharing the top level builder as shown below.
+
+All the clients created from a shared `ServiceBusClientBuilder` instance will share the same connection to the Service Bus namespace.
+
+
```

Comment:
> Hence, for the use cases with many clients, expecting continuous consumption and sending of messages with reasonable load, it would be beneficial to use dedicated builder instance per client or load balance the clients among multiple builder instances. 

Maybe we can replace it with something like:

If your application throughput to one ServiceBus endpoint is high for each instance (in terms of number of sent/received messages or payload size), consider creating multiple clients. For example, you can create an individual  client per each queue or topic and subscription name. In case of extremely high load you might also need to create multiple instances of clients for a single queue or subscription and load balance between them.

You can detect IO thread bottlenecks by profiling you application and checking how busy IO threads are (Service Bus SDK uses `reactor-executor-*` naming pattern for the IO threads).